### PR TITLE
[eccodes] Version 2.36

### PR DESCRIPTION
**Pull request by barche**

### CODE REVIEW

File: E/eccodes/build_tarballs.jl [hunk 1]
```diff
@@ -3,11 +3,11 @@
 using BinaryBuilder, Pkg
 
 name = "eccodes"
-version = v"2.28.0"
+version = v"2.36.0"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://confluence.ecmwf.int/download/attachments/45757960/eccodes-$version-Source.tar.gz", "2831347b1517af9ebd70dd3cad88ae818a8448d4e6c8671aa728617e73431cd5"),
+    ArchiveSource("https://confluence.ecmwf.int/download/attachments/45757960/eccodes-2.36.0-Source.tar.gz", "da74143a64b2beea25ea27c63875bc8ec294e69e5bd0887802040eb04151d79a"),
```
Comment (giordano) [inline E/eccodes/build_tarballs.jl:10]: Why $version was removed?

File: E/eccodes/build_tarballs.jl [hunk 2]
```diff
@@ -41,6 +41,10 @@ install_license ../LICENSE
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = supported_platforms()
+# 32 bit platforms are not supported by eccodes
```
Comment (giordano) [inline E/eccodes/build_tarballs.jl:44]: nbits(p) == 32 should be clearer than listing the platforms

File: E/eccodes/build_tarballs.jl (hunk 1 above)
Comment (barche) [reply E/eccodes/build_tarballs.jl]: Ah, forgot to fix up after copy/pasting the URL, thanks!

**Comment by barche** [comment]
Something is wrong on Windows, the tests for GRIB.jl with the new binary fail with error:

```
GRIB: Test Failed at C:\Users\bartj\.julia\packages\GRIB\y8dyK\test\runtests.jl:242
  Expression: msg["name"] == "Echo Top"
   Evaluated: "unknown" == "Echo Top"

Stacktrace:
 [1] macro expansion
   @ C:\Users\bartj\.julia\juliaup\julia-1.10.4+0.x64.w64.mingw32\share\julia\stdlib\v1.10\Test\src\Test.jl:672 [inlined]
 [2] (::var"#21#42")(f::GribFile)
   @ Main C:\Users\bartj\.julia\packages\GRIB\y8dyK\test\runtests.jl:242
 [3] GribFile(f::var"#21#42", filename::String; mode::String)
   @ GRIB C:\Users\bartj\.julia\packages\GRIB\y8dyK\src\gribfile.jl:54
 [4] GribFile(f::Function, filename::String)
   @ GRIB C:\Users\bartj\.julia\packages\GRIB\y8dyK\src\gribfile.jl:51
 [5] macro expansion
   @ C:\Users\bartj\.julia\packages\GRIB\y8dyK\test\runtests.jl:240 [inlined]
 [6] macro expansion
   @ C:\Users\bartj\.julia\juliaup\julia-1.10.4+0.x64.w64.mingw32\share\julia\stdlib\v1.10\Test\src\Test.jl:1577 [inlined]
 [7] top-level scope
   @ C:\Users\bartj\.julia\packages\GRIB\y8dyK\test\runtests.jl:7
```

**Comment by barche** [comment]
New patch based on the one from MSYS2 fixes the Windows problem.

**Pull request merged**